### PR TITLE
feat(recovery): [CON-1425] Allow skipping nodes during certification downloads

### PR DESCRIPTION
**Pull request by dist1ll**
This PR adds a way to manually skip nodes during certification downloads when rsync fails.

Background: During recoveries, we should download and merge the certification pools of as many nodes on the subnet as possible. Sometimes there are connection issues to individual nodes, and the tool gets stuck trying to download the node’s pool. In those cases we may want to skip the problematic nodes.

### CODE REVIEW

File: rs/recovery/src/file_sync_helper.rs [hunk 1]
```diff
@@ -85,12 +86,25 @@ pub fn rsync_with_retries(
...
-            Err(e) => warn!(logger, "Rsync failed: {:?}, retrying...", e),
+            Err(e) => {
+                warn!(logger, "Rsync failed: {:?}", e);
+                if auto_retry {
+                    // In non-interactive cases, we wait a short while
+                    // before re-trying rsync.
+                    info!(logger, "Retrying in 10 seconds...");
+                    std::thread::sleep(std::time::Duration::from_secs(10));
+                } else if !consent_given("Do you want to retry the  download for this node?") {
+                    return Err(RecoveryError::UnexpectedError(
```
Comment (kpop-dfinity) [inline rs/recovery/src/file_sync_helper.rs:97]: Maybe we should introduce a new error variant? To me `UnexpectedError` doesn't feel right here.

Not sure what a good name would be though

File: rs/recovery/src/file_sync_helper.rs [hunk 2]
```diff
@@ -85,12 +86,25 @@ pub fn rsync_with_retries(
...
             key_file,
         ) {
-            Err(e) => warn!(logger, "Rsync failed: {:?}, retrying...", e),
+            Err(e) => {
+                warn!(logger, "Rsync failed: {:?}", e);
+                if auto_retry {
+                    // In non-interactive cases, we wait a short while
+                    // before re-trying rsync.
+                    info!(logger, "Retrying in 10 seconds...");
+                    std::thread::sleep(std::time::Duration::from_secs(10));
```
Comment (kpop-dfinity) [inline rs/recovery/src/file_sync_helper.rs:95]: nit: we already import `std::time::Duration`

```suggestion
                    std::thread::sleep(Duration::from_secs(10));
```

**Review by kpop-dfinity — APPROVED** [review]

File: rs/recovery/src/file_sync_helper.rs (hunk 2 above)
Comment (dist1ll) [reply rs/recovery/src/file_sync_helper.rs]: done

File: rs/recovery/src/file_sync_helper.rs (hunk 1 above)
Comment (dist1ll) [reply rs/recovery/src/file_sync_helper.rs]: What do you think about a simple `RsyncFailed` variant? Updated the code.

**Pull request merged**